### PR TITLE
Refactor Follow

### PR DESCRIPTION
**Pull request by vuducdat**

### CODE REVIEW

File: app/controllers/followed_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class FollowedController < ApplicationController
+  before_action :logged_in_user, only: [ :index]
+
+  def index
+    @title = t("title.follower")
```
Comment (leminhtuan2015) [inline app/controllers/followed_controller.rb]: ( )

File: app/controllers/followed_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class FollowedController < ApplicationController
+  before_action :logged_in_user, only: [ :index]
```
Comment (leminhtuan2015) [inline app/controllers/followed_controller.rb]: [ ]

File: app/controllers/following_controller.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+class FollowingController < ApplicationController
+  before_action :logged_in_user, only: [ :index]
```
Comment (leminhtuan2015) [inline app/controllers/following_controller.rb]: [ ]

File: app/controllers/relationships_controller.rb [hunk 4]
```diff
@@ -4,10 +4,17 @@ class RelationshipsController < ApplicationController
   def create
     user = User.find params[:followed_id]
     current_user.follow user
+    respond_to do |format|
+      format.html { redirect_to @user }
```
Comment (leminhtuan2015) [inline app/controllers/relationships_controller.rb]: Convebntion

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,9 +1,19 @@
...
   before_action :correct_user, only: [:edit, :update]
 
   def show
-    @user = User.find params[:id]
+    @user = User.find params[:id] 
+  end
+  
+  def index
+    @user = User.find params[:user_id]
+    if !params[:type].blank? && ["following", "followers"].include?(params[:type])
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: move `["following", "followers"]` to Setting

File: app/controllers/relationships_controller.rb [hunk 6]
```diff
@@ -4,10 +4,27 @@ class RelationshipsController < ApplicationController
   def create
     user = User.find params[:followed_id]
     current_user.follow user
+    respond_to do |format|
+      format.html {redirect_to @user}
+      format.js
+    end
+  end
+
+  def index
```
Comment (dieunb) [inline app/controllers/relationships_controller.rb]: order action should follow `index, new, show, create, edit, update, destroy`

File: config/settings.yml [hunk 7]
```diff
@@ -12,3 +12,6 @@ per_page: 10
 avatar_size_default: 80
 avatar_size: 30
 avatar_size_user: 50
+follows:
```
Comment (dieunb) [inline config/settings.yml]: naming not good.

File: app/controllers/relationships_controller.rb [hunk 8]
```diff
@@ -1,13 +1,31 @@
...
   before_action :logged_in_user
 
+  def index
+    @user = User.find params[:user_id]
+    if !params[:type].blank? && Settings.naming_for_action_follow.include?
+      (params[:type])
+      @title = params[:type]
+      @users = @user.send(params[:type])
+      @users = @users.paginate page: params[:page]
+      render :index
```
Comment (dieunb) [inline app/controllers/relationships_controller.rb]: remove

File: app/controllers/relationships_controller.rb [hunk 9]
```diff
@@ -1,13 +1,31 @@
 class RelationshipsController < ApplicationController
   before_action :logged_in_user
 
+  def index
+    @user = User.find params[:user_id]
+    if !params[:type].blank? && Settings.naming_for_action_follow.include?
+      (params[:type])
+      @title = params[:type]
+      @users = @user.send(params[:type])
```
Comment (dieunb) [inline app/controllers/relationships_controller.rb]: remove `()`

Comment (dieunb) [inline app/controllers/relationships_controller.rb]:
```
+      @users = @user.send(params[:type]).paginate page: params[:page]
```

**Pull request merged**